src/cobra/io/mat.py

```diff
@@ -284,139 +673,148 @@ def from_mat_struct(
+    rxn_lbs = [-inf if np.isinf(x) and x < 0 else x for x in rxn_lbs]
+    rxn_ubs = _cell_to_float_list(m["ub"][0, 0])
+    rxn_ubs = [-inf if np.isinf(x) and x > 0 else x for x in rxn_ubs]
+    rxn_gene_rules, rxn_names, rxn_subsystems = None, None, None
+    try:
+        rxn_gene_rules = _cell_to_str_list(m["grRules"][0, 0], "")
+    except (IndexError, ValueError):
+        # TODO: use custom cobra exception to handle exception
+        pass
+    try:
+        rxn_names = _cell_to_str_list(m["rxnNames"][0, 0], "")
+    except (IndexError, ValueError):
+        # TODO: use custom cobra exception to handle exception
+        pass
+    try:
+        # RECON3.0 mat has an array within an array for subsystems.
+        # If we find a model that has multiple subsytems per reaction, this should be
+        # modified
+        if isinstance(m["subSystems"][0, 0][0][0][0], np.ndarray):
+            rxn_subsystems = [
+                _each_cell[0][0][0][0] if _each_cell else None
+                for _each_cell in m["subSystems"][0, 0]
+            ]
+        # Other matlab files seem normal.
+        else:
+            rxn_subsystems = _cell_to_str_list(m["subSystems"][0, 0])
+    except (IndexError, ValueError):
+        # TODO: use custom cobra exception to handle exception
+        pass
+    for i in range(len(rxn_ids)):
+        new_reaction = Reaction(
+            id=rxn_ids[i], lower_bound=rxn_lbs[i], upper_bound=rxn_ubs[i]
+        )
+        if rxn_names:
+            new_reaction.name = rxn_names[i]
+        if rxn_subsystems:
+            new_reaction.subsystem = rxn_subsystems[i]
+        if rxn_gene_rules:
+            new_reaction.gene_reaction_rule = rxn_gene_rules[i]
         new_reactions.append(new_reaction)
-    model.add_reactions(new_reactions)
-    set_objective(model, coefficients)
-
-    coo = scipy_sparse.coo_matrix(m["S"][0, 0])
-    for i, j, v in zip(coo.row, coo.col, coo.data):
-        model.reactions[j].add_metabolites({model.metabolites[i]: v})
-    return model
-
-
-def _check(result: Dict[str, str]) -> None:
-    """Ensure success of a `pymatbridge` operation.
-
-    Parameters
-    ----------
-    result : dict
-        The dictionary obtained from `pymatbridge` with keys as 'message',
-        'result', and 'success'.
-
-    Raises
-    ------
-    RuntimeError
-        If setting variable via `pymatbridge` fails.
-
-    """
-    if result["success"] is not True:
-        # TODO: verify if key 'content' is valid as docs don't state about it
-        raise RuntimeError(result["content"]["stdout"])
+    mat_parse_annotations(new_reactions, m, _d_replace=D_REACTION)
+    mat_parse_notes(new_reactions, m, _d_replace=D_REACTION_NOTES)
 
+    csc = scipy_sparse.csc_matrix(m["S"][0, 0])
+    for i in range(csc.shape[1]):
+        stoic_dict = {
+            model.metabolites[j]: csc[j, i] for j in csc.getcol(i).nonzero()[0]
+        }
+        new_reactions[i].add_metabolites(stoic_dict)
 
-def model_to_pymatbridge(
```

Comment:
Good for me to deactivate that since we have been looking for someone to test this for 2+ years now and nobody has stepped up. We should keep in mind that this requires a new release though since it removes existing functionality.

Reply:
Okay